providers/build.info

```diff
@@ -1,63 +1,47 @@
-# linking.
-#
-# The non-legacy implementations (libimplementations) must be made FIPS
-# agnostic as much as possible, as well as the common building blocks
-# (libcommon).  The legacy implementations (liblegacy) will never be
-# part of the FIPS provider.
-#
-# If there is anything that isn't FIPS agnostic, it should be set aside
-# in its own source file, which is then included directly into other
-# static libraries geared for FIPS and non-FIPS providers, and built
-# separately.
-#
-# libcommon.a           Contains common building blocks, potentially
-#                       needed both by non-legacy and legacy code.
-#
-# libimplementations.a  Contains all non-legacy implementations.
-# liblegacy.a           Contains all legacy implementations.
-#
-# libfips.a             Contains all things needed to support
-#                       FIPS implementations, such as code from
-#                       crypto/ and object files that contain
-#                       FIPS-specific code.  FIPS_MODULE is defined
-#                       for this library.  The FIPS module uses
-#                       this.
-# libnonfips.a          Corresponds to libfips.a, but built with
-#                       FIPS_MODULE undefined.  The default and legacy
-#                       providers use this.
-#
-# This is how different provider modules should be linked:
-#
-# FIPS:
-#     -o fips.so {object files...} libimplementations.a libcommon.a libfips.a
-# Non-FIPS:
-#     -o module.so {object files...} libimplementations.a libcommon.a libnonfips.a
-#
-# It is crucial that code that checks for the FIPS_MODULE macro end up in
-# libfips.a and libnonfips.a, never in libcommon.a.
-# It is crucial that such code is written so libfips.a and libnonfips.a doesn't
-# end up depending on libimplementations.a or libcommon.a.
-# It is crucial that such code is written so libcommon.a doesn't end up
-# depending on libimplementations.a.
-#
-# Code in providers/implementations/ should be written in such a way that the
-# OSSL_DISPATCH arrays (and preferably the majority of the actual code) ends
-# up in either libimplementations.a or liblegacy.a.
-# If need be, write an abstraction layer in separate source files and make them
-# libfips.a / libnonfips.a sources.
+# libcommon.a           Contains common building blocks and other routines,
+#                       potentially needed both by any of our providers.
+#
+# libfips.a             Contains all FIPS compilations of the algorithm
+#                       implementations, which is primarly code from
+#                       providers/implementations, but also code that it
+#                       happens to share with libcrypto, i.e. from crypto/.
+#                       FIPS_MODULE is defined for object files placed in
+#                       this library, and it's exclusively used by the FIPS
+#                       provider module.
+#
+# liblegacy.a           Corresponds to libfips.a, but built with FIPS_MODULE
```

Comment:
'Corresponds to' seems confusing to me..

Reply:
Hmmm, right.

Reply:
I've rewritten the explanation for libfips.a, liblegacy.a and libdefault.a, please have a look.